pyrene/src/utils/DateUtils.ts

```diff
@@ -0,0 +1,264 @@
+
+/**
+ * Returns the Date Object of provided timestamp in UTC
+ */
+export const convertToUTCtime = (datetime: string | number, timezone: string): Date => zonedTimeToUtc(datetime, timezone);
+
+/**
+ * Returns the timestamp of a point in time in the future relative to now
+ *
+ * @param {Duration} duration
+ */
+export const getFutureDate = (duration: Duration): number => add(getCurrentJsDateObject(), duration).valueOf();
+
+/**
+ * Returns the timestamp of a point in time in the past relative to now
+ *
+ * @param {Duration} duration
+ */
+export const getPastDate = (duration: Duration): number => sub(getCurrentJsDateObject(), duration).valueOf();
+
+// isExists uses 0 indexed month numbers
+/**
+ * Checks if a timetype is valid
+ * @param {DateType} date
+ */
+export const isValidDate = (date?: DateType): boolean => {
+  if (date) {
+    return isExists(date.year, date.month - 1, date.day);
+  }
+  return false;
+};
+
+/**
+ * Checks if a timetype is valid
+ * @param {TimeType} time
+ */
+export const isValidTime = (time?: TimeType): boolean => {
+  if (time && time.hours >= 0 && time.hours <= 23 && time.minutes >= 0 && time.minutes <= 59) {
+    return true;
+  }
+  return false;
+};
+
+/** Checks if a timezone string is valid
+* @param {string} timezone
+*/
+export const isValidTimeZone = (timezone: string): boolean => {
+  try {
+    utcToZonedTime(getCurrentJsDateObject(), timezone);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const zeroFill = (num: string, length: number): string => (num.toString().padStart(length, '0'));
+
+export const convertDateTypeToString = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${year}-${month}-${day}`;
+};
+
+export const convertTimeTypeToString = (time: TimeType): string => {
+  const hours = zeroFill(time.hours.toString(), 2);
+  const minutes = zeroFill(time.minutes.toString(), 2);
+
+  return `${hours}:${minutes}`;
+};
+
+export const dateTypeToStandardEUDateFormat = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${day}.${month}.${year}`;
+};
+
+export const standardEUDateFormat = (date: Date): string => format(date, 'dd.MM.yyyy');
+
+export const standardEUTimeFormat = (date: Date): string => format(date, 'HH:mm');
+
+export const getDateTypeFromddmmyyyyWithSep = (str: string): DateType | undefined => {
+  if (str.length === 10 && allowedSeparatorCheck(str.charAt(2)) && allowedSeparatorCheck(str.charAt(5))) {
+    const date = { day: +str.substr(0, 2), month: +str.substr(3, 2), year: +str.substr(6) };
+    if (!Number.isNaN(date.day) && !Number.isNaN(date.month) && !Number.isNaN(date.year)) {
+      return date;
```

Comment:
Two remarks about the function `getDateTypeFromddmmyyyyWithSep`
1. This function `parses` a date in form of a string
2. This function depends on another function called `allowedSeparatorCheck`.

**My advices**
1. Use an established `parser` for that, like this

```
import { parse, getDay, getMonth, getYear, isValid } from 'date-fns';

const getDateTypeFromddmmyyyyWithSep = (dateString: string): DateType | undefined => {
    const parsed = parse(dateString, 'dd.MM.yyyy hh:mm', new Date());
    return isValid(parsed) ? { day: getDay(parsed), month: getMonth(parsed), year: getYear(parsed) } : undefined;
};

```

2. Use the `Dependency Injection` pattern to pass a required function in the body of the main function.

Reply:
This function has been removed